src/core/generators/axios.ts

```diff
@@ -147,14 +148,14 @@ export const generateAxiosHeader = ({
   : never;\n\n`
     : ''
 }
-  ${!noFunction ? `export const ${title} = () => {\n` : ''}`;
+  ${!noFunction ? `export const ${title} = ({axios: localAxios = axios}: {axios?: AxiosInstance} = {}) => {\n` : ''}`;
```

Comment:
I think that we also need to check if there is a global mutator here. Because if that's the case we don't use the axios instance